geocube/geo_utils/geobox.py

```diff
@@ -153,7 +145,9 @@ def __init__(self, output_crs, resolution, align, geom, like):
 
         """
         self.output_crs = output_crs
-        self.resolution = resolution
+        self.resolution = (
+            resolution if not isinstance(resolution, tuple) else Resolution(*resolution)
```

Comment:
I recommend using `resyx_` constructor function here instead of `Resolution` class, this communicates more clearly what order is expected to be present in the tuple.
```suggestion
            resolution if not isinstance(resolution, tuple) else resyx_(*resolution)
```

https://odc-geo.readthedocs.io/en/latest/_api/odc.geo.resyx_.html#odc.geo.resyx_